File: app/src/main/java/me/foolishchow/androidplugins/MainActivity.java
package me.foolishchow.androidplugins;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import java.util.List;

import me.foolishchow.android.annotation.IntentParam;


public class MainActivity extends AppCompatActivity {

    @IntentParam
    List<Integer> ints;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //Class<MainActivity$ShadowLifeCycleDelegate> classes =
        //        MainActivity$ShadowLifeCycleDelegate.class;
        //mKernalViewBinding.mNavTitle.setText("sdasdasdasd");
        //setContentView(R.layout.activity_main);
        //MainActivity$ShadowLifeCycleDelegate mainActivity$ShadowLifeCycleDelegate = new MainActivity$ShadowLifeCycleDelegate();
        //mainActivity$ShadowLifeCycleDelegate.restoreInstanceState();
    }


}

File: app/src/main/java/me/foolishchow/autoparamdemo/MainActivity.kt
package me.foolishchow.autoparamdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import me.foolishchow.android.annotation.IntentParam

open class MainActivity : AppCompatActivity() {
    @JvmField
    @IntentParam
    protected var ints: List<Int>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val controller = navController
        //NavigationManager.attach(controller, R.navigation.main)
        //val navDestinations = MainActivity(controller)
        //val inflate = controller!!.navInflater.inflate(R.navigation.main)
        //val displayName = inflate.navigatorName
    }

    private val navController: NavController?
        get() {
            try {
                val loginNavHost = supportFragmentManager.findFragmentByTag("login_nav_host") as NavHostFragment?
                return loginNavHost?.navController
            } catch (ignored: Exception) {
            }
            return null
        }
}